annif/backend/stwfsa.py

```diff
@@ -0,0 +1,121 @@
+import os
+from stwfsapy.predictor import StwfsapyPredictor
+from annif.exception import NotInitializedException, NotSupportedException
+from annif.suggestion import ListSuggestionResult, SubjectSuggestion
+from . import backend
+from annif.util import boolean
+
+
+_KEY_CONCEPT_TYPE_URI = 'concept_type_uri'
+_KEY_SUBTHESAURUS_TYPE_URI = 'sub_thesaurus_type_uri'
+_KEY_THESAURUS_RELATION_TYPE_URI = 'thesaurus_relation_type_uri'
+_KEY_THESAURUS_RELATION_IS_SPECIALISATION = (
+    'thesaurus_relation_is_specialisation')
+_KEY_REMOVE_DEPRECATED = 'remove_deprecated'
+_KEY_HANDLE_TITLE_CASE = 'handle_title_case'
+_KEY_EXTRACT_UPPER_CASE_FROM_BRACES = 'extract_upper_case_from_braces'
+_KEY_EXTRACT_ANY_CASE_FROM_BRACES = 'extract_any_case_from_braces'
+_KEY_EXPAND_AMPERSAND_WITH_SPACES = 'expand_ampersand_with_spaces'
+_KEY_EXPAND_ABBREVIATION_WITH_PUNCTUATION = (
+    'expand_abbreviation_with_punctuation')
+_KEY_SIMPLE_ENGLISH_PLURAL_RULES = 'simple_english_plural_rules'
+_KEY_INPUT_LIMIT = 'input_limit'
+
+
+class StwfsaBackend(backend.AnnifBackend):
+
+    name = "stwfsa"
+    needs_subject_index = True
+
+    STWFSA_PARAMETERS = {
+        _KEY_CONCEPT_TYPE_URI: str,
+        _KEY_SUBTHESAURUS_TYPE_URI: str,
+        _KEY_THESAURUS_RELATION_TYPE_URI: str,
+        _KEY_THESAURUS_RELATION_IS_SPECIALISATION: boolean,
+        _KEY_REMOVE_DEPRECATED: boolean,
+        _KEY_HANDLE_TITLE_CASE: boolean,
+        _KEY_EXTRACT_UPPER_CASE_FROM_BRACES: boolean,
+        _KEY_EXTRACT_ANY_CASE_FROM_BRACES: boolean,
+        _KEY_EXPAND_AMPERSAND_WITH_SPACES: boolean,
+        _KEY_EXPAND_ABBREVIATION_WITH_PUNCTUATION: boolean,
+        _KEY_SIMPLE_ENGLISH_PLURAL_RULES: boolean,
+        _KEY_INPUT_LIMIT: int,
+    }
+
+    DEFAULT_PARAMETERS = {
+        _KEY_SUBTHESAURUS_TYPE_URI: '',
+        _KEY_THESAURUS_RELATION_TYPE_URI: '',
+        _KEY_THESAURUS_RELATION_IS_SPECIALISATION: False,
+        _KEY_REMOVE_DEPRECATED: True,
+        _KEY_HANDLE_TITLE_CASE: True,
+        _KEY_EXTRACT_UPPER_CASE_FROM_BRACES: True,
+        _KEY_EXTRACT_ANY_CASE_FROM_BRACES: False,
+        _KEY_EXPAND_AMPERSAND_WITH_SPACES: True,
+        _KEY_EXPAND_ABBREVIATION_WITH_PUNCTUATION: True,
+        _KEY_SIMPLE_ENGLISH_PLURAL_RULES: False,
+        _KEY_INPUT_LIMIT: 0,
+    }
+
+    MODEL_FILE = 'stwfsa_predictor.zip'
+
+    _model = None
+
+    def initialize(self):
+        if self._model is None:
+            path = os.path.join(self.datadir, self.MODEL_FILE)
+            self.debug(f'Loading STWFSA model from {path}.')
+            if os.path.exists(path):
+                self._model = StwfsapyPredictor.load(path)
+                self.debug('Loaded model.')
+            else:
+                raise NotInitializedException(
```

Comment:
This line 71 was not covered by the tests. I suggest that you add a test that checks this case, similar to [test_omikuji_suggest_no_vectorizer](https://github.com/NatLibFi/Annif/blob/267cc8518e392530a543995cef6df7c0caf6273b/tests/test_backend_omikuji.py#L28)